index.js

```diff
@@ -0,0 +1,57 @@
+const express = require("express");
+const bcrypt = require("bcryptjs");
+const db = require("./dbHelpers");
+
+const server = express();
+const PORT = 4045;
+
+server.use(express.json());
+
+server.get("/", (req, res) => {
+  res.send("Yep, this is the server!");
+});
+
+server.post("/api/register", (req, res) => {
+  const user = req.body;
+  user.password = bcrypt.hashSync(user.password, 14);
+  if (user.username && user.password) {
+    db.insertUser(user)
+      .then(id => {
+        res.status(201).json({ message: `User created with the id of ${id}` });
+      })
+      .catch(err => {
+        res.json(err);
+      });
+  } else {
+    res.status(400).json({ message: "Please enter a username and password" });
+  }
+});
+
+server.post("/api/login", (req, res) => {
+  const user = req.body;
+  db.findByUsername(user.username)
+    .then(dbUser => {
+      if (dbUser[0] && bcrypt.compareSync(user.password, dbUser[0].password)) {
```

Comment:
This credential check looks good Rob 👍 you're checking for the existence of the user and for a matching password hash.